### PR TITLE
feat(stacking): add getSecondsUntilStackingDeadline helper method

### LINKED ISSUE #1514: getSecondsUntilNextCycle does not use perpare phase
### What version of Stacks.js are you using?

6.5.4

### Describe the bug

getSecondsUntilNextCycle returns the seconds until the start of the reward phase, not of the prepare phase.
The prepare phase is important for stackers because stx locked after the prepare phase do not earn rewards during the upcoming reward phase.

#### How to reproduce
Check getSecondsUntilNextCycle() and compare to blocks_until_prepare_phase of e.g. https://api.hiro.so/v2/pox

#### Expected behavior

Return estimated seconds until start of next prepare phase 

### Additional context
Used in lockstacks.com/pool-admin

image

### PR DESCRIPTION
**Pull request by janniks**
> This PR was published to npm with the version `6.5.5-pr.231b5a8.0`
> e.g. `npm install @stacks/common@6.5.5-pr.231b5a8.0 --save-exact`

@friedger does this make more sense? 🙏

### CODE REVIEW

File: packages/stacking/src/index.ts [hunk 1]
```diff
@@ -504,6 +505,23 @@ export class StackingClient {
     );
   }
 
+  /**
+   * Get number of seconds until the end of the stacking deadline.
+   * This is the estimated time stackers have to submit their stacking
+   * transactions to be included in the upcoming reward cycle.
```
Comment (friedger) [inline packages/stacking/src/index.ts:504]: The function returns a negative value during the prepare phase (after the stacking deadline and before users can stack again for the next cycle).

File: packages/stacking/tests/stacking.test.ts [hunk 2]
```diff
@@ -1084,3 +1085,24 @@ test('client operations with contract principal stacker', () => {
...
+  });
+
+  seconds = await client.getSecondsUntilStackingDeadline();
+  expect(seconds).toBe(0); // this time we are in the prepare phase
```
Comment (friedger) [inline packages/stacking/tests/stacking.test.ts:1107]: Maybe add a test with a negative number for `blocks_until_prepare_phase` to document the behaviour during prepare phase.

**Review by friedger — APPROVED** [review]
Looks good, added suggestions to improve documentation.
